src/Form/Traits/AllowedLinkClassesTrait.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+namespace SilverStripe\LinkField\Form\Traits;
+
+use SilverStripe\Core\Injector\Injector;
+use SilverStripe\LinkField\Models\Link;
+use SilverStripe\LinkField\Services\LinkTypeService;
+
+/**
+ * Trait to manage which Link type can be added to LinkField form field.
+ * This trait is used in LinkField and MultiLinkField classes.
+ */
+trait AllowedLinkClassesTrait
+{
+    private $allowed_types = [];
+
+    /**
+     * Set allowed types for LinkField
+     */
+    public function setAllowedTypes(array $types): static
+    {
+        if (empty($types) || !$this->validateTypes($types)) {
+            $this->allowed_types = $this->genarateAllowedTypes();
+        } else {
+            $this->allowed_types = $this->genarateAllowedTypes($types);
+        }
+
+        return $this;
+    }
+
+    /**
+     * Get allowed types for LinkField
+     */
+    public function getAllowedTypes(): array
+    {
+        return $this->allowed_types;
+    }
+
+    /**
+     * Validate types that they are subclasses of Link
+     */
+    private function validateTypes(array $types): bool
+    {
+        $validClasses = [];
+        foreach ($types as $type) {
+            if (is_subclass_of($type, Link::class)) {
+                $validClasses[] = $type;
+            }
+        }
+
+        return count($validClasses) > 0;
+    }
+
+    /**
+     * The method returns an associational array converted to a JSON string,
+     * of available link types with additional parameters necessary
+     * for full-fledged work on the client side.
+     * @throws InvalidArgumentException
+     */
+    public function getTypesProps(): string
+    {
+        $types = [];
+        $typeDefinitions = $this->genarateAllowedTypes();
+
+        foreach ($typeDefinitions as $key => $class) {
+            $types[$key] = Injector::inst()->get($class);
+        }
+
+        $typesList = [];
+
+        array_map(function (Link $type, string $key) use (&$typesList) {
+            $typesList[$key] = [
+                'key' => $key,
+                'title' => $type->LinkTypeTile(),
+                'handlerName' => $type->LinkTypeHandlerName(),
+            ];
+        }, $types, array_keys($types));
```

Comment:
I think this is more concise.

```suggestion
$typesList = [];
        $typeDefinitions = $this->genarateAllowedTypes();

        foreach ($typeDefinitions as $key => $class) {
            /** Link $type */
            $type = Injector::inst()->get($class);
            $typesList[$key] = [
                'key' => $key,
                'title' => $type->LinkTypeTile(),
                'handlerName' => $type->LinkTypeHandlerName(),
            ];
        }
```

Reply:
Updated